### PR TITLE
Auditv2 unzipping path fixes

### PR DESCRIPTION
**Pull request by escherize**
Instead of appending the path to the plugins dir, this updates unzip-file to take a function from entry-name to the path to put the file.

e.g.

If the entry is at `"/Users/user/dev/metabase/metabase/resources/instance_analytics/databases/Internal Metabase Database/Internal Metabase Database.yaml"`, we want it to end up at `"/Users/user/dev/metabase/metabase/plugins/instance_analytics/databases/Internal Metabase Database/Internal Metabase Database.yaml"`, not `"/Users/user/dev/metabase/metabase//Users/user/dev/metabase/metabase/plugins/instance_analytics/databases/Internal Metabase Database/Internal Metabase Database.yaml"`.

This should fix rows 0-23 of this handy table:
```
| :row | :plugins-dir | :ed |  :db | :jar | :extraction-mode |
|------+--------------+-----+------+------+------------------|
|    0 |      vanilla |  ee |   pg |   no |            unzip |
|    1 |      vanilla |  ee |   pg |   no |             copy |
|    2 |      vanilla |  ee |   h2 |   no |            unzip |
|    3 |      vanilla |  ee |   h2 |   no |             copy |
|    4 |      vanilla |  ee | msql |   no |            unzip |
|    5 |      vanilla |  ee | msql |   no |             copy |
|    6 |      vanilla |  ee |   pg |  jar |            unzip |
|    7 |      vanilla |  ee |   pg |  jar |             copy |
|    8 |      vanilla |  ee |   h2 |  jar |            unzip |
|    9 |      vanilla |  ee |   h2 |  jar |             copy |
|   10 |      vanilla |  ee | msql |  jar |            unzip |
|   11 |      vanilla |  ee | msql |  jar |             copy |
|   12 |      vanilla | oss |   pg |   no |            unzip |
|   13 |      vanilla | oss |   pg |   no |             copy |
|   14 |      vanilla | oss |   h2 |   no |            unzip |
|   15 |      vanilla | oss |   h2 |   no |             copy |
|   16 |      vanilla | oss | msql |   no |            unzip |
|   17 |      vanilla | oss | msql |   no |             copy |
|   18 |      vanilla | oss |   pg |  jar |            unzip |
|   19 |      vanilla | oss |   pg |  jar |             copy |
|   20 |      vanilla | oss |   h2 |  jar |            unzip |
|   21 |      vanilla | oss |   h2 |  jar |             copy |
|   22 |      vanilla | oss | msql |  jar |            unzip |
|   23 |      vanilla | oss | msql |  jar |             copy |
|   24 |   overridden |  ee |   pg |   no |            unzip |
|   25 |   overridden |  ee |   pg |   no |             copy |
|   26 |   overridden |  ee |   h2 |   no |            unzip |
|   27 |   overridden |  ee |   h2 |   no |             copy |
|   28 |   overridden |  ee | msql |   no |            unzip |
|   29 |   overridden |  ee | msql |   no |             copy |
|   30 |   overridden |  ee |   pg |  jar |            unzip |
|   31 |   overridden |  ee |   pg |  jar |             copy |
|   32 |   overridden |  ee |   h2 |  jar |            unzip |
|   33 |   overridden |  ee |   h2 |  jar |             copy |
|   34 |   overridden |  ee | msql |  jar |            unzip |
|   35 |   overridden |  ee | msql |  jar |             copy |
|   36 |   overridden | oss |   pg |   no |            unzip |
|   37 |   overridden | oss |   pg |   no |             copy |
|   38 |   overridden | oss |   h2 |   no |            unzip |
|   39 |   overridden | oss |   h2 |   no |             copy |
|   40 |   overridden | oss | msql |   no |            unzip |
|   41 |   overridden | oss | msql |   no |             copy |
|   42 |   overridden | oss |   pg |  jar |            unzip |
|   43 |   overridden | oss |   pg |  jar |             copy |
|   44 |   overridden | oss |   h2 |  jar |            unzip |
|   45 |   overridden | oss |   h2 |  jar |             copy |
|   46 |   overridden | oss | msql |  jar |            unzip |
|   47 |   overridden | oss | msql |  jar |             copy |
```

There's some more work to do for handling overriding the plugins directory. (which should have us working for the rest of the combinations in the table.)

### CODE REVIEW

**Comment by escherize** [comment]
when running in dev, these paths are relative paths. and when running from a jar, the paths are absolute paths. so the previous approach of appending onto the plugins directory did NOT WORK IN JARs. but, it does now.

File: enterprise/backend/src/metabase_enterprise/audit_db.clj [hunk 1]
```diff
@@ -129,7 +132,11 @@
   (cond
     zip-resource
     (do (log/info "Unzipping instance_analytics to " plugin-dir "...")
-        (u.files/unzip-file analytics-zip-resource plugin-dir)
+        (u.files/unzip-file analytics-zip-resource
+                            (fn [entry-name]
+                              (str/replace entry-name
+                                           "resources/instance_analytics/"
+                                           "plugins/instance_analytics/")))
```
Comment (noahmoss) [inline enterprise/backend/src/metabase_enterprise/audit_db.clj:139]: probably shouldn't be hard-coding `plugins` here either

**Review by noahmoss — APPROVED** [review]
👍 with one comment

File: enterprise/backend/src/metabase_enterprise/audit_db.clj (hunk 1 above)
Comment (escherize) [reply enterprise/backend/src/metabase_enterprise/audit_db.clj:139]: yeah, even the `"resources/instance_analytics/"` doesn't feel quite right. I'll go figure out a better abstraction tomorrow.

**Pull request merged**